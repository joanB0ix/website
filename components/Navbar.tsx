import { SunIcon } from "@heroicons/react/24/solid";

export default function Navbar() {
  return (
    <nav className="h-20 flex items-center justify-between">
      <div>
        <h1 className="font-bold">Joan Boix Ã€valos</h1>
      </div>
      <div>
        <SunIcon className="h-5 w-5 mb-1 mr-5" data-testid="sun-icon" />
      </div>
    </nav>
  );
}
