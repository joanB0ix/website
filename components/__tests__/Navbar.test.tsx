import { render, screen } from "@testing-library/react";
import Navbar from "../Navbar";

describe("Navbar Component", () => {
  it("renders correctly", () => {
    render(<Navbar />);

    const navbarTitle = screen.getByText("Joan Boix Ã€valos");
    const sunIcon = screen.getByTestId("sun-icon");

    expect(navbarTitle).toBeInTheDocument();
    expect(sunIcon).toBeInTheDocument();
    expect(sunIcon).toHaveClass("h-5 w-5 mb-1 mr-5");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
