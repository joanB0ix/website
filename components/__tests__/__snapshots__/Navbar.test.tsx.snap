// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="h-20 flex items-center justify-between"
  >
    <div>
      <h1
        class="font-bold"
      >
        Joan Boix Ã€valos
      </h1>
    </div>
    <div>
      <svg
        aria-hidden="true"
        class="h-5 w-5 mb-1 mr-5"
        data-slot="icon"
        data-testid="sun-icon"
        fill="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2.25a.75.75 0 0 1 .75.75v2.25a.75.75 0 0 1-1.5 0V3a.75.75 0 0 1 .75-.75ZM7.5 12a4.5 4.5 0 1 1 9 0 4.5 4.5 0 0 1-9 0ZM18.894 6.166a.75.75 0 0 0-1.06-1.06l-1.591 1.59a.75.75 0 1 0 1.06 1.061l1.591-1.59ZM21.75 12a.75.75 0 0 1-.75.75h-2.25a.75.75 0 0 1 0-1.5H21a.75.75 0 0 1 .75.75ZM17.834 18.894a.75.75 0 0 0 1.06-1.06l-1.59-1.591a.75.75 0 1 0-1.061 1.06l1.59 1.591ZM12 18a.75.75 0 0 1 .75.75V21a.75.75 0 0 1-1.5 0v-2.25A.75.75 0 0 1 12 18ZM7.758 17.303a.75.75 0 0 0-1.061-1.06l-1.591 1.59a.75.75 0 0 0 1.06 1.061l1.591-1.59ZM6 12a.75.75 0 0 1-.75.75H3a.75.75 0 0 1 0-1.5h2.25A.75.75 0 0 1 6 12ZM6.697 7.757a.75.75 0 0 0 1.06-1.06l-1.59-1.591a.75.75 0 0 0-1.061 1.06l1.59 1.591Z"
        />
      </svg>
    </div>
  </nav>
</DocumentFragment>
`;
